### PR TITLE
[chore] Add release checklist

### PR DESCRIPTION
**Pull request by dvdplm**
Include changes from the v0.5.1 release﻿prep:
- add a release checklist
- fix markdown lints
- keep crates versions in sync: 0.5.1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,26 +55,28 @@ taking advantage of the better reading performance of sequential data access. Ga
...
-## Polkadot Archive
+### Polkadot Archive
+
 - [Changed] Config file is now optional. Can configure polkadot archive entirely through environment variables.
   - the environment variables that need to be set are `CHAIN_DATA_DB` and `DATABASE_URL`.
 - [Changed] Polkadot archive will archive `polkadot` by default if the `--chain` CLI option is not passed.
 - [Changed] remove `rpc_url` from the polkadot-archive TOML configuration file
 - [Changed] All options in config file apart from `db_url`.
 
+## [v0.4.0]
```
Comment (insipx) [inline CHANGELOG.md:66]:
```suggestion
## **[v0.4.0]**
```
 
For consistency

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+## **[Unreleased]**
 
-# [v0.5.0]
+## **[v0.5.1]**
+
+- [Added] Release checklist
+- [Changed] Pinned to `substrate`/`polkadot` release v0.8.30
+- [Changed] Keeping the `substrate-archive` and `substrate-archive-backend` crates's versions aligned.
+
+## [v0.5.0]
```
Comment (insipx) [inline CHANGELOG.md:11]:
```suggestion
## **[v0.5.0]**
```

**Review by insipx — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,19 +33,19 @@
 - [Added] Two new options to the configuration relative to state tracing:
   - `targets` for specifying runtime targets to trace in WASM
   - `folder` where WASM blobs with tracing enabled are kept.
-  - More on state-tracing here: https://github.com/paritytech/substrate-archive/wiki/6.)-State-Tracing-&-Balance-Reconciliation  
+  - More on state-tracing [here](https://github.com/paritytech/substrate-archive/wiki/6.)-State-Tracing-&-Balance-Reconciliation)
```
Comment (niklasad1) [inline CHANGELOG.md:36]: this needs to fixed again ^^

Comment (insipx) [inline CHANGELOG.md:36]:
```suggestion
  - More on state-tracing [here](https://github.com/paritytech/substrate-archive/wiki/6.\)-State-Tracing-&-Balance-Reconciliation)
```

**Review by niklasad1 — APPROVED** [review]

**Pull request merged**